runtime/src/main/cpp/Runtime.cpp

```diff
@@ -82,6 +83,7 @@ RuntimeState* initRuntime() {
   bool firstRuntime = atomicAdd(&aliveRuntimesCount, 1) == 1;
   // Keep global variables in state as well.
   if (firstRuntime) {
+    isMainThread = 1;
```

Comment:
What if Kotlin runtime gets initialized on non-main thread first?

Reply:
It is impossible with this definition of the main thread - main thread *is* the first thread which initialized the runtime. It's up to platform-specific code to ensure that Kotlin/Native main thread matches to GUI thread or whatever is the the main thread in other senses.

Reply:
> It's up to platform-specific code to ensure that Kotlin/Native main thread matches to GUI thread

How do you suggest to implement this on iOS then?

Reply:
Call properly in launcher for direct interop, let's discuss how to do for reverse interop. My gut feeling is that somewhere near https://github.com/JetBrains/kotlin-native/blob/9b2fc6c6dddfa9e749eea21dce8f3fb79a34b1a4/runtime/src/main/cpp/ObjCExport.mm#L418 may help.